<Type Name="LinqDataSourceValidationException" FullName="System.Web.UI.WebControls.LinqDataSourceValidationException">
  <Metadata><Meta Name="ms.openlocfilehash" Value="62766df0ebc45e194067654c3d1a68e83473c5be" /><Meta Name="ms.sourcegitcommit" Value="756d085f27705e86604f1bba5f2086ee23761acf" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="es-ES" /><Meta Name="ms.lasthandoff" Value="01/30/2019" /><Meta Name="ms.locfileid" Value="55312223" /></Metadata><TypeSignature Language="C#" Value="public class LinqDataSourceValidationException : Exception, System.Web.DynamicData.IDynamicValidatorException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit LinqDataSourceValidationException extends System.Exception implements class System.Runtime.Serialization.ISerializable, class System.Web.DynamicData.IDynamicValidatorException" />
  <TypeSignature Language="DocId" Value="T:System.Web.UI.WebControls.LinqDataSourceValidationException" />
  <TypeSignature Language="VB.NET" Value="Public Class LinqDataSourceValidationException&#xA;Inherits Exception&#xA;Implements IDynamicValidatorException" />
  <TypeSignature Language="C++ CLI" Value="public ref class LinqDataSourceValidationException : Exception, System::Web::DynamicData::IDynamicValidatorException" />
  <TypeSignature Language="F#" Value="type LinqDataSourceValidationException = class&#xA;    inherit Exception&#xA;    interface IDynamicValidatorException&#xA;    interface ISerializable" />
  <AssemblyInfo>
    <AssemblyName>System.Web.Extensions</AssemblyName>
    <AssemblyVersion>3.5.0.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Exception</BaseTypeName>
  </Base>
  <Interfaces>
    <Interface>
      <InterfaceName>System.Runtime.Serialization.ISerializable</InterfaceName>
    </Interface>
    <Interface>
      <InterfaceName>System.Web.DynamicData.IDynamicValidatorException</InterfaceName>
    </Interface>
  </Interfaces>
  <Attributes>
    <Attribute>
      <AttributeName>System.Serializable</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary><span data-ttu-id="0a6cd-101">Describe una excepción que se produjo durante la validación de valores nuevos o modificados, antes de que un control <see cref="T:System.Web.UI.WebControls.LinqDataSource" /> insertara, actualizara o eliminara los valores.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-101">Describes an exception that occurred during validation of new or modified values before values are inserted, updated, or deleted by a <see cref="T:System.Web.UI.WebControls.LinqDataSource" /> control.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="0a6cd-102">La <xref:System.Web.UI.WebControls.LinqDataSourceValidationException> clase contiene excepciones de establecedor de conversión y la propiedad de tipo.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-102">The <xref:System.Web.UI.WebControls.LinqDataSourceValidationException> class contains type conversion and property setter exceptions.</span></span> <span data-ttu-id="0a6cd-103">Todas las excepciones que se producen durante la validación se encuentran en el <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> colección.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-103">All of the exceptions that are thrown during validation are contained in the <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> collection.</span></span> <span data-ttu-id="0a6cd-104">Puede recuperar cada excepción de validación al recorrer el <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> colección.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-104">You can retrieve each validation exception by iterating through the <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> collection.</span></span>  
  
 <span data-ttu-id="0a6cd-105">Normalmente controlan las excepciones en el <xref:System.Web.UI.WebControls.LinqDataSource.Updating>, <xref:System.Web.UI.WebControls.LinqDataSource.Deleting>, y <xref:System.Web.UI.WebControls.LinqDataSource.Inserting> eventos.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-105">You typically handle exceptions in the <xref:System.Web.UI.WebControls.LinqDataSource.Updating>, <xref:System.Web.UI.WebControls.LinqDataSource.Deleting>, and <xref:System.Web.UI.WebControls.LinqDataSource.Inserting> events.</span></span> <span data-ttu-id="0a6cd-106">Si se produce una excepción de validación, el `Exception` propiedad de la <xref:System.Web.UI.WebControls.LinqDataSourceUpdateEventArgs>, <xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs>, o <xref:System.Web.UI.WebControls.LinqDataSourceInsertEventArgs> objeto contiene un <xref:System.Web.UI.WebControls.LinqDataSourceValidationException> objeto.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-106">If a validation exception is thrown, the `Exception` property of the <xref:System.Web.UI.WebControls.LinqDataSourceUpdateEventArgs>, <xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs>, or <xref:System.Web.UI.WebControls.LinqDataSourceInsertEventArgs> object contains a <xref:System.Web.UI.WebControls.LinqDataSourceValidationException> object.</span></span> <span data-ttu-id="0a6cd-107">Si se produce ninguna excepción, el `Exception` contiene la propiedad `null`.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-107">If no exception is thrown, the `Exception` property contains `null`.</span></span>  
  
 <span data-ttu-id="0a6cd-108">Para controlar las excepciones de validación, cree un controlador para el <xref:System.Web.UI.WebControls.LinqDataSource.Updating>, <xref:System.Web.UI.WebControls.LinqDataSource.Deleting>, o <xref:System.Web.UI.WebControls.LinqDataSource.Inserting> eventos.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-108">To handle validation exceptions, create a handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Updating>, <xref:System.Web.UI.WebControls.LinqDataSource.Deleting>, or <xref:System.Web.UI.WebControls.LinqDataSource.Inserting> event.</span></span> <span data-ttu-id="0a6cd-109">En el controlador de eventos, compruebe si el `Exception` propiedad de la clase de argumento de evento contiene un valor distinto de `null`.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-109">In the event handler, check whether the `Exception` property of the event argument class contains a value other than `null`.</span></span> <span data-ttu-id="0a6cd-110">Si el `Exception` propiedad no es null, se produjeron uno o más excepciones de validación y se puede recuperar cada excepción de validación en el <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> propiedad.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-110">If the `Exception` property is not null, one or more validation exceptions were thrown and you can retrieve each validation exception in the <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> property.</span></span>  
  
 <span data-ttu-id="0a6cd-111">Puede producirse una excepción de validación si un valor no coincide con el tipo de la propiedad.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-111">A validation exception can occur if a value does not match the type of the property.</span></span> <span data-ttu-id="0a6cd-112">Por ejemplo, si se intenta actualizar una propiedad de entero mediante el uso de caracteres no numéricos, hace una excepción de validación.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-112">For example, trying to update an integer property by using non-numeric characters causes a validation exception.</span></span> <span data-ttu-id="0a6cd-113">Una clase LINQ to SQL también puede contener los criterios de validación personalizada que asegúrese de que la propiedad contiene un valor que está dentro de un intervalo esperado o un patrón.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-113">A LINQ to SQL class can also contain customized validation criteria that make sure that the property contains a value that is within an expected range or pattern.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="0a6cd-114">El ejemplo siguiente muestra un controlador de eventos para el <xref:System.Web.UI.WebControls.LinqDataSource.Updating> eventos.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-114">The following example shows an event handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Updating> event.</span></span> <span data-ttu-id="0a6cd-115">Muestra los mensajes de excepción de validación utilizando un <xref:System.Web.UI.WebControls.Label> control.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-115">It displays any validation exception messages by using a <xref:System.Web.UI.WebControls.Label> control.</span></span>  
  
```vb  
Protected Sub LinqDataSource_Updating(ByVal sender As Object, _  
        ByVal e As LinqDataSourceUpdateEventArgs)  
    If (e.Exception IsNot Nothing) Then  
        For Each innerException As KeyValuePair(Of String, Exception) _  
               In e.Exception.InnerExceptions  
            Label1.Text &= innerException.Key & ": " & _  
                innerException.Value.Message & "<br />"  
        Next  
        e.ExceptionHandled = True  
    End If  
End Sub  
```  
  
```csharp  
protected void LinqDataSource_Updating(object sender,   
        LinqDataSourceUpdateEventArgs e)  
{  
    if (e.Exception != null)  
    {  
        foreach (KeyValuePair<string, Exception> innerException in   
             e.Exception.InnerExceptions)  
        {  
        Label1.Text += innerException.Key + ": " +   
            innerException.Value.Message + "<br />";  
        }  
        e.ExceptionHandled = true;  
    }  
}  
```  
  
 ]]></format>
    </remarks>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Web.Extensions</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary><span data-ttu-id="0a6cd-116">Inicializa una nueva instancia de la clase <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" />.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-116">Initializes a new instance of the <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> class.</span></span></summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public LinqDataSourceValidationException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.UI.WebControls.LinqDataSourceValidationException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; LinqDataSourceValidationException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web.Extensions</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="0a6cd-117">Inicializa una nueva instancia de la clase <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> con un mensaje proporcionado por el sistema que describe el error.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-117">Initializes a new instance of the <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> class with a system-supplied message that describes the error.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="0a6cd-118">El mensaje proporcionado por el sistema que describe el error tiene en cuenta la referencia cultural actual del sistema.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-118">The system-supplied message that describes the error takes into account the current system culture.</span></span>  
  
 <span data-ttu-id="0a6cd-119">En la siguiente tabla se muestran los valores iniciales de las propiedades de una instancia de la clase <xref:System.Web.UI.WebControls.LinqDataSourceValidationException>.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-119">The following table shows the initial property values for an instance of the <xref:System.Web.UI.WebControls.LinqDataSourceValidationException> class.</span></span>  
  
|<span data-ttu-id="0a6cd-120">Propiedad.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-120">Property</span></span>|<span data-ttu-id="0a6cd-121">Valor</span><span class="sxs-lookup"><span data-stu-id="0a6cd-121">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="0a6cd-122">`null`.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-122">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="0a6cd-123">Una descripción traducida proporcionada por el sistema.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-123">A system-supplied localized description.</span></span>|  
  
 ]]></format>
        </remarks>
        <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="0a6cd-124">Controlar y generar excepciones</span><span class="sxs-lookup"><span data-stu-id="0a6cd-124">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public LinqDataSourceValidationException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.UI.WebControls.LinqDataSourceValidationException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; LinqDataSourceValidationException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new System.Web.UI.WebControls.LinqDataSourceValidationException : string -&gt; System.Web.UI.WebControls.LinqDataSourceValidationException" Usage="new System.Web.UI.WebControls.LinqDataSourceValidationException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web.Extensions</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="0a6cd-125">Mensaje que describe el error.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-125">The message that describes the error.</span></span></param>
        <summary><span data-ttu-id="0a6cd-126">Inicializa una nueva instancia de la clase <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> con un mensaje especificado que describe el error.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-126">Initializes a new instance of the <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> class with a specified message that describes the error.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="0a6cd-127">En la siguiente tabla se muestran los valores iniciales de las propiedades de una instancia de la clase <xref:System.Web.UI.WebControls.LinqDataSourceValidationException>.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-127">The following table shows the initial property values for an instance of the <xref:System.Web.UI.WebControls.LinqDataSourceValidationException> class.</span></span>  
  
|<span data-ttu-id="0a6cd-128">Propiedad.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-128">Property</span></span>|<span data-ttu-id="0a6cd-129">Valor</span><span class="sxs-lookup"><span data-stu-id="0a6cd-129">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="0a6cd-130">`null`.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-130">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="0a6cd-131">La cadena de mensaje de error especificada en `message`.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-131">The error message string specified in `message`.</span></span>|  
  
 ]]></format>
        </remarks>
        <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="0a6cd-132">Controlar y generar excepciones</span><span class="sxs-lookup"><span data-stu-id="0a6cd-132">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected LinqDataSourceValidationException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.UI.WebControls.LinqDataSourceValidationException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; LinqDataSourceValidationException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="new System.Web.UI.WebControls.LinqDataSourceValidationException : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; System.Web.UI.WebControls.LinqDataSourceValidationException" Usage="new System.Web.UI.WebControls.LinqDataSourceValidationException (info, context)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web.Extensions</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info"><span data-ttu-id="0a6cd-133">Objeto que contiene los datos del objeto serializados.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-133">The object that holds the serialized object data.</span></span></param>
        <param name="context"><span data-ttu-id="0a6cd-134">Información de contexto acerca del origen o del destino.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-134">Context information about the source or destination.</span></span></param>
        <summary><span data-ttu-id="0a6cd-135">Inicializa una nueva instancia de la clase <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> con datos serializados.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-135">Initializes a new instance of the <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> class with serialized data.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="0a6cd-136">Este constructor se llama durante la deserialización para reconstituir el objeto de excepción que se transmite a través de una secuencia.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-136">This constructor is called during deserialization to reconstitute the exception object that is transmitted over a stream.</span></span>  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public LinqDataSourceValidationException (string message, System.Collections.Generic.IDictionary&lt;string,Exception&gt; innerExceptions);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Collections.Generic.IDictionary`2&lt;string, class System.Exception&gt; innerExceptions) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.UI.WebControls.LinqDataSourceValidationException.#ctor(System.String,System.Collections.Generic.IDictionary{System.String,System.Exception})" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerExceptions As IDictionary(Of String, Exception))" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; LinqDataSourceValidationException(System::String ^ message, System::Collections::Generic::IDictionary&lt;System::String ^, Exception ^&gt; ^ innerExceptions);" />
      <MemberSignature Language="F#" Value="new System.Web.UI.WebControls.LinqDataSourceValidationException : string * System.Collections.Generic.IDictionary&lt;string, Exception&gt; -&gt; System.Web.UI.WebControls.LinqDataSourceValidationException" Usage="new System.Web.UI.WebControls.LinqDataSourceValidationException (message, innerExceptions)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web.Extensions</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerExceptions" Type="System.Collections.Generic.IDictionary&lt;System.String,System.Exception&gt;" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="0a6cd-137">Mensaje que describe el error.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-137">The message that describes the error.</span></span></param>
        <param name="innerExceptions"><span data-ttu-id="0a6cd-138">Colección de excepciones de validación.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-138">The collection of validation exceptions.</span></span></param>
        <summary><span data-ttu-id="0a6cd-139">Inicializa una nueva instancia de la clase <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> con un mensaje especificado que describe el error y una colección de referencias a excepciones internas.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-139">Initializes a new instance of the <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> class with a specified message that describes the error, and a collection of references to inner exceptions.</span></span></summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public LinqDataSourceValidationException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.UI.WebControls.LinqDataSourceValidationException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; LinqDataSourceValidationException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new System.Web.UI.WebControls.LinqDataSourceValidationException : string * Exception -&gt; System.Web.UI.WebControls.LinqDataSourceValidationException" Usage="new System.Web.UI.WebControls.LinqDataSourceValidationException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web.Extensions</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="0a6cd-140">Mensaje que describe el error.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-140">The message that describes the error.</span></span></param>
        <param name="innerException"><span data-ttu-id="0a6cd-141">Excepción que es la causa de la excepción actual.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-141">The exception that is the cause of the current exception.</span></span></param>
        <summary><span data-ttu-id="0a6cd-142">Inicializa una nueva instancia de la clase <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> con un mensaje de error especificado y una referencia a la excepción interna.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-142">Initializes a new instance of the <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> class with a specified error message and a reference to the inner exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="0a6cd-143">Normalmente, no utilizará este constructor porque las excepciones de validación se almacenan en el <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> propiedad, no el <xref:System.Exception.InnerException%2A> propiedad.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-143">You typically will not use this constructor because validation exceptions are stored in the <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> property, not the <xref:System.Exception.InnerException%2A> property.</span></span>  
  
 ]]></format>
        </remarks>
        <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="0a6cd-144">Controlar y generar excepciones</span><span class="sxs-lookup"><span data-stu-id="0a6cd-144">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
    <Member MemberName="GetObjectData">
      <MemberSignature Language="C#" Value="public override void GetObjectData (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig virtual instance void GetObjectData(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.UI.WebControls.LinqDataSourceValidationException.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Public Overrides Sub GetObjectData (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; override void GetObjectData(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="override this.GetObjectData : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; unit" Usage="linqDataSourceValidationException.GetObjectData (info, context)" />
      <MemberType>Method</MemberType>
      <Implements>
        <InterfaceMember>M:System.Runtime.Serialization.ISerializable.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)</InterfaceMember>
      </Implements>
      <AssemblyInfo>
        <AssemblyName>System.Web.Extensions</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8">
          <AttributeName>System.Security.SecurityCritical</AttributeName>
        </Attribute>
      </Attributes>
      <ReturnValue>
        <ReturnType>System.Void</ReturnType>
      </ReturnValue>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info"><span data-ttu-id="0a6cd-145">Objeto que contiene los datos serializados del objeto para la excepción que se inicia.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-145">The object that holds the serialized object data for the exception that is being thrown.</span></span></param>
        <param name="context"><span data-ttu-id="0a6cd-146">Información de contexto acerca del origen o del destino.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-146">Context information about the source or destination.</span></span></param>
        <summary><span data-ttu-id="0a6cd-147">Obtiene información sobre la excepción y la agrega al objeto <see cref="T:System.Runtime.Serialization.SerializationInfo" />.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-147">Gets information about the exception and adds it to the <see cref="T:System.Runtime.Serialization.SerializationInfo" /> object.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="0a6cd-148">El <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.GetObjectData%2A> método agrega información sobre la excepción a la <xref:System.Runtime.Serialization.SerializationInfo> objeto destinado para la serialización.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-148">The <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.GetObjectData%2A> method adds information about the exception to the <xref:System.Runtime.Serialization.SerializationInfo> object that is targeted for serialization.</span></span> <span data-ttu-id="0a6cd-149">Durante la deserialización, la excepción se reconstituye a partir del <xref:System.Runtime.Serialization.SerializationInfo> que se transmite a través de la secuencia.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-149">During deserialization, the exception is reconstituted from the <xref:System.Runtime.Serialization.SerializationInfo> that is transmitted over the stream.</span></span>  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName="InnerExceptions">
      <MemberSignature Language="C#" Value="public System.Collections.Generic.IDictionary&lt;string,Exception&gt; InnerExceptions { get; }" />
      <MemberSignature Language="ILAsm" Value=".property instance class System.Collections.Generic.IDictionary`2&lt;string, class System.Exception&gt; InnerExceptions" />
      <MemberSignature Language="DocId" Value="P:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions" />
      <MemberSignature Language="VB.NET" Value="Public ReadOnly Property InnerExceptions As IDictionary(Of String, Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; property System::Collections::Generic::IDictionary&lt;System::String ^, Exception ^&gt; ^ InnerExceptions { System::Collections::Generic::IDictionary&lt;System::String ^, Exception ^&gt; ^ get(); };" />
      <MemberSignature Language="F#" Value="member this.InnerExceptions : System.Collections.Generic.IDictionary&lt;string, Exception&gt;" Usage="System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions" />
      <MemberType>Property</MemberType>
      <Implements>
        <InterfaceMember>P:System.Web.DynamicData.IDynamicValidatorException.InnerExceptions</InterfaceMember>
      </Implements>
      <AssemblyInfo>
        <AssemblyName>System.Web.Extensions</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Collections.Generic.IDictionary&lt;System.String,System.Exception&gt;</ReturnType>
      </ReturnValue>
      <Docs>
        <summary><span data-ttu-id="0a6cd-150">Obtiene una o varias excepciones que produjeron durante la validación de datos nuevos o modificados.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-150">Gets one or more exceptions that occurred when new or modified data was being validated.</span></span></summary>
        <value><span data-ttu-id="0a6cd-151">Colección que contiene las excepciones.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-151">A collection that contains the exceptions.</span></span></value>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="0a6cd-152">El <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> colección contiene todas las excepciones de validación que se produjeron durante la validación de datos antes de una actualización, inserción o la operación de eliminación.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-152">The <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> collection contains all the validation exceptions that were thrown during data validation before an update, insert, or delete operation.</span></span> <span data-ttu-id="0a6cd-153">Puede producirse una excepción de validación si un valor no coincide con el tipo de la propiedad.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-153">A validation exception can occur if a value does not match the type of the property.</span></span> <span data-ttu-id="0a6cd-154">Por ejemplo, si se intenta actualizar una propiedad de entero mediante el uso de caracteres no numéricos, se produce una excepción de validación.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-154">For example, if you try to update an integer property by using non-numeric characters, a validation exception is thrown.</span></span> <span data-ttu-id="0a6cd-155">Una clase LINQ to SQL también puede contener los criterios de validación personalizada que asegúrese de que la propiedad contiene un valor que está dentro de un intervalo esperado o un patrón.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-155">A LINQ to SQL class can also contain customized validation criteria that make sure that the property contains a value that is within an expected range or pattern.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="0a6cd-156">El ejemplo siguiente muestra un controlador de eventos para el <xref:System.Web.UI.WebControls.LinqDataSource.Updating> eventos.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-156">The following example shows an event handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Updating> event.</span></span> <span data-ttu-id="0a6cd-157">Muestra los mensajes de excepción de validación utilizando un <xref:System.Web.UI.WebControls.Label> control.</span><span class="sxs-lookup"><span data-stu-id="0a6cd-157">It displays any validation exception messages by using a <xref:System.Web.UI.WebControls.Label> control.</span></span>  
  
```vb  
Protected Sub LinqDataSource_Updating(ByVal sender As Object, _  
        ByVal e As LinqDataSourceUpdateEventArgs)  
    If (e.Exception IsNot Nothing) Then  
        For Each innerException As KeyValuePair(Of String, Exception) _  
                In e.Exception.InnerExceptions  
          Label1.Text &= innerException.Key & ": " & _   
                  innerException.Value.Message & "<br />"  
        Next  
        e.ExceptionHandled = True  
    End If  
End Sub  
```  
  
```csharp  
protected void LinqDataSource_Updating(object sender,   
        LinqDataSourceUpdateEventArgs e)   
{  
    if (e.Exception != null)  
    {  
        foreach (KeyValuePair<string, Exception> innerException in   
            e.Exception.InnerExceptions)  
        {  
            Label1.Text += innerException.Key + ": " +   
                innerException.Message + "<br />";  
        }  
        e.ExceptionHandled = true;  
    }  
}  
```  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
  </Members>
</Type>